cli/src/main/java/com/devonfw/tools/ide/commandlet/UpgradeSettingsCommandlet.java

```diff
@@ -181,4 +187,43 @@ private static void updatePropertiesLegacyEdition(EnvironmentVariablesProperties
       environmentVariables.remove(legacyEditionVariable);
     }
   }
+
+  private void cleanupLegacyProperties() {
+    this.context.info("Cleaning up legacy properties...");
+
+    Path rootDirectory = Paths.get(System.getProperty("user.dir"));
+    try {
+      Files.walk(rootDirectory)
+          .filter(path -> path.getFileName().toString().equals("IDEasy.properties"))
+          .forEach(filePath -> {
+            try {
+              updateProperties(filePath);
+            } catch (IOException e) {
+              this.context.warning("Error processing IDEasy.properties at " + filePath);
+            }
+          });
+    } catch (IOException e) {
+      this.context.warning("Error walking the file tree to find IDEasy.properties.");
+    }
+  }
+
+  private void updateProperties(Path filePath) throws IOException {
```

Comment:
Maybe rename to make more clear what this is doing:
```suggestion
  private void updateRepositoryPropertiesFile(Path filePath) throws IOException {
```